### PR TITLE
(DOCSP-26621): Refactor Flutter Realm Database IA

### PR DESCRIPTION
**Pull request by mongodben**
## Pull Request Info

Refactor Flutter Realm Database IA to better align w new standard

note to reviewer: i consistently make redirect-related mistakes (including 2 fixed by this PR). pls be mindful of the redirects changes to make sure i didn't mess anything up 😄 

### Jira

- https://jira.mongodb.org/browse/DOCSP-26621

### Staged Changes (Requires MongoDB Corp SSO)

- [Flutter SDK (whole section)](https://docs-mongodbcom-staging.corp.mongodb.com/realm/docsworker-xlarge/DOCSP-26621/sdk/flutter/)

### Review Guidelines

[REVIEWING.md](https://github.com/mongodb/docs-realm/blob/master/REVIEWING.md)

### CODE REVIEW

File: source/sdk/flutter/realm-files.txt [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+- Provide a schema version or migration block when making schema changes
+
+.. _flutter-realm-file:
+
+Realm Files
+-----------
+
+Realm Database stores a binary encoded version of every object and type in a
+realm in a single ``.realm`` file. The file is located at :ref:`a specific
+path <find-the-default-realm-path>` that you can define when you open the realm.
```
Comment (dacharyc) [inline source/sdk/flutter/realm-files.txt:45]: The way this is worded, we talk about how you can define the path when you open the realm. I expected that to link to [Customize Default Configuration](https://www.mongodb.com/docs/realm/sdk/flutter/realm-database/open-and-close-a-realm/#customize-default-configuration) where we talk about how you can define a path for the realm file. Instead, this links to the Studio docs for "finding" the Realm File Path. We might want to reword this, or move the link to the "define" section?

(Also, I just noticed we haven't updated that section of the Studio docs in ages. We should probably add info there for how to find the Realm file path for Flutter, and probably also for Kotlin and maybe CPP soon. I'll make a ticket for that.)

File: source/sdk/flutter/realm-database.txt [hunk 2]
```diff
@@ -42,72 +30,4 @@ mobile database:
 Realm Database is an alternative to `SQLite <https://www.sqlite.org/index.html>`__
 and :apple:`Core Data <documentation/coredata>`.
 
-Features
```
Comment (dacharyc) [inline source/sdk/flutter/realm-database.txt:45]: It looks like lines 14-31 above appear twice on the staged page. Maybe they're also in the include you've added below?

File: config/redirects [hunk 3]
```diff
@@ -851,8 +851,16 @@ raw: ${prefix}/tutorial -> ${appServicesBase}/get-started/
 
 # Flutter SDK docs expansion
 raw: ${prefix}/sdk/flutter/realm-database/data-types -> ${appServicesBase}/sdk/flutter/model-data/data-types/
-raw: ${prefix}/sdk/flutter/realm-database/define-realm-object-schema -> ${appServicesBase}/sdk/flutter/define-realm-object-schema/
-raw: ${prefix}/sdk/flutter/realm-database/update-realm-object-schema -> ${appServicesBase}/sdk/flutter/update-realm-object-schema/
+raw: ${prefix}/sdk/flutter/realm-database/define-realm-object-schema -> ${appServicesBase}/sdk/flutter/model-data/define-realm-object-schema/
```
Comment (dacharyc) [inline config/redirects:854]: These redirects all look good EXCEPT they're all using `${appServicesBase}` and I think you want just `${base}` here as we're not actually moving these pages to App Services. (I think that also applies to the prior redirect at ln 853, which you haven't modified in this PR but seems to be doing the same thing. If I reviewed that PR and missed it then, apologies!)

```suggestion
raw: ${prefix}/sdk/flutter/realm-database/define-realm-object-schema -> ${base}/sdk/flutter/model-data/define-realm-object-schema/
```

**Review by dacharyc — APPROVED** [review]
Overall, looks good! Just one Q about where we link something, and I think the redirects are all pointing to the wrong base - should be good to go after fixing those.

File: config/redirects (hunk 3 above)
Comment (mongodben) [reply config/redirects:854]: thx! good catch

File: source/sdk/flutter/realm-database.txt (hunk 2 above)
Comment (mongodben) [reply source/sdk/flutter/realm-database.txt:45]: yep, thx for catching. fixing

File: source/sdk/flutter/realm-files.txt (hunk 1 above)
Comment (mongodben) [reply source/sdk/flutter/realm-files.txt:45]: yea makes sense. will update here. btw i lifted this text from swift docs.

will change there too if i can find a good ref

Comment (mongodben) [reply source/sdk/flutter/realm-files.txt:45]: couldn't find a relevant place to link in swift. this section seems to be the closest thing - https://www.mongodb.com/docs/realm/sdk/swift/realm-files/configure-and-open-a-realm/#default-realm

don't know if you want to add anything there, but wanted to put on your radar.

**Pull request merged**